File: SRAlbum/ShowTableViewCell.m
//
//  ShowTableViewCell.m
//  T
//
//  Created by 施峰磊 on 2019/3/18.
//  Copyright © 2019 施峰磊. All rights reserved.
//

#import "ShowTableViewCell.h"
#import "FLAnimatedImageView.h"
#import "FLAnimatedImage.h"

@interface ShowTableViewCell()
@property (weak, nonatomic) IBOutlet FLAnimatedImageView *image_view;
@property (weak, nonatomic) IBOutlet NSLayoutConstraint *height;

@end

@implementation ShowTableViewCell

- (void)awakeFromNib {
    [super awakeFromNib];
    // Initialization code
}

- (void)setImage:(UIImage *)image{
    CGSize imageSize;
    if ([image isKindOfClass:[NSData class]]) {
        FLAnimatedImage *animatedImage = [FLAnimatedImage animatedImageWithGIFData:(NSData *)image];
        self.image_view.animatedImage = animatedImage;
        imageSize = animatedImage.size;
    }else{
        self.image_view.image = image;
        imageSize = image.size;
    }
    self.height.constant = imageSize.height*[UIScreen mainScreen].bounds.size.width/imageSize.width;
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
    [super setSelected:selected animated:animated];

    // Configure the view for the selected state
}

@end

File: SRAlbum/ShowTableViewCell.swift
//
//  ShowTableViewCell.swift
//  SRAlbum
//
//  Created by 施峰磊 on 2020/2/21.
//  Copyright © 2020 施峰磊. All rights reserved.
//

import UIKit

class ShowTableViewCell: UITableViewCell {
    @IBOutlet weak var showImageView: UIImageView!
    
    
    var showImage:UIImage?{
        didSet{
            self.showImageView.image = self.showImage;
        }
    }
    

    override func awakeFromNib() {
        super.awakeFromNib()
        // Initialization code
    }

    override func setSelected(_ selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)

        // Configure the view for the selected state
    }
    
}
